src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -301,30 +378,73 @@ public int getIdleMinutes() {
         return (idleMinutes != null) ? idleMinutes : 0;
     }
 
+    @DataBoundSetter
+    public void setIdleMinutes(Integer idleMinutes) {
+        this.idleMinutes = (idleMinutes != null) ? idleMinutes : 0;
+    }
+
     public int getMaxSize() {
-        return maxSize;
+        return Math.max(1, maxSize);
```

Comment:
This wouldn't work for the case when `maxSize` is less than 1 and `minSize` is more than 1. 
If max size can be equal to min size, shouldn't this be `Math.max(minSize, maxSize)`? 

Can we add a test for this case?

Reply:
By using the setters in the deprecated constructor, can't we simply return the value in getters and move validations/default setting to setters?

Reply:
It is fine for `getMaxSize` since its validated by `setMaxSize` so there is no case where `maxSize < 1` when this code is called.

Yes, I read that we should just put the deprecated tag on the old constructor but based on the Kubernetes Cloud it seems like it is ok to actually change it too.